relay-server/src/actors/processor.rs

```diff
@@ -2434,11 +2439,115 @@ mod tests {
+        let rate_limited_org = 0;
+        let not_ratelimited_org = 1;
+
+        let message = {
+            let mut scopes = BTreeMap::<Scoping, ProjectMetrics>::new();
+
+            let quota = Quota {
+                id: Some("testing".into()),
+                categories: vec![DataCategory::MetricBucket].into(),
+                scope: relay_quotas::QuotaScope::Organization,
+                scope_id: Some(rate_limited_org.to_string()), // we will ratelimit buckets with scope org-id == 1
+                limit: Some(0),
+                window: None,
+                reason_code: Some(ReasonCode::new("test")),
+            };
+
+            let bucket = Bucket {
+                name: "d:transactions/bar".to_string(),
+                value: BucketValue::Counter(1.0),
+                timestamp: UnixTimestamp::now(),
+                tags: Default::default(),
+                width: 10,
+            };
+
+            let scoping_by_org_id = |org_id: u64| Scoping {
+                organization_id: org_id,
+                project_id: ProjectId::new(21),
+                project_key: ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap(),
+                key_id: Some(17),
+            };
+
+            let project_state = {
+                let mut config = ProjectConfig::default();
+                config.quotas.push(quota);
+
+                let mut project_state = ProjectState::allowed();
+                project_state.config = config;
+                Arc::new(project_state)
+            };
+
+            let project_metrics = ProjectMetrics {
+                buckets: vec![bucket],
+                project_state,
+            };
+
+            scopes.insert(scoping_by_org_id(rate_limited_org), project_metrics.clone());
+            scopes.insert(scoping_by_org_id(not_ratelimited_org), project_metrics);
+
+            EncodeMetrics { scopes }
+        };
+
+        // ensure the order of the map while iterating is as expected.
+        let mut iter = message.scopes.keys();
+        assert_eq!(iter.next().unwrap().organization_id, rate_limited_org);
+        assert_eq!(iter.next().unwrap().organization_id, not_ratelimited_org);
+        assert!(iter.next().is_none());
+
+        let config = {
+            let config_json = serde_json::json!({
+                "processing": {
+                    "enabled": true,
+                    "kafka_config": [],
+                    "redis": {
+                        "server": std::env::var("RELAY_REDIS_URL").unwrap_or_else(|_| "redis://127.0.0.1:6379".to_owned()),
+                    }
+                }
+            });
+            Config::from_json_value(config_json).unwrap()
+        };
+
+        let (store, handle) = {
+            let f = |org_ids: &mut Vec<u64>, msg: Store| {
+                let org_id = match msg {
+                    Store::Metrics(x) => x.scoping.organization_id,
+                    Store::Envelope(_) => panic!(),
```

Comment:
Could we set a message in this panic? If this test detects a regression in the future, a nice error message can help scope down the cause.